src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -60,139 +53,110 @@
       }
     }
-
-    InitCodeGeneratorContext initCodeContext =
-        InitCodeGeneratorContext.newBuilder()
-            .symbolTable(table)
-            .flattenedFields(fields)
-            .initStructure(initFieldStructure)
-            .initObjectName(Name.from("request"))
-            .initObjectType(method.getInputType())
-            .build();
-    InitCodeGenerator generator = new InitCodeGenerator();
-    InitCode initCode = generator.generate(initCodeContext);
-
-    return buildInitCodeView(methodContext, initCode);
+    InitCodeNode rootNode =
+        InitCodeNode.createSpecItemTree(
+            SpecItemParserContext.newBuilder()
+                .table(table)
+                .valueGenerator(valueGenerator)
+                .rootObjectType(context.getMethod().getOutputType())
+                .initValueConfigMap(createInitValueMap(context))
+                .subTrees(createMockResponseInitSubTrees(context))
+                .initFieldSet(primitiveFields)
+                .suggestedName(Name.from("expected_response"))
+                .build());
+    return buildInitCodeViewRequestObject(context, rootNode);
   }
 
-  private Map<String, Object> createSmokeTestInitMap(
-      List<String> smokeTestInitFields, MethodTransformerContext context) {
-    Map<String, Object> initFieldStructure =
-        FieldStructureParser.parseFields(smokeTestInitFields, createInitValueMap(context));
-    return initFieldStructure;
-  }
-
-  private InitCodeView buildInitCodeView(MethodTransformerContext context, InitCode initCode) {
-    ImportTypeTransformer importTypeTransformer = new ImportTypeTransformer();
-    ModelTypeTable typeTable = context.getTypeTable();
-    SurfaceNamer namer = context.getNamer();
-
-    // Initialize the type table with the apiClassName since each sample will be using the
-    // apiClass.
-    typeTable.getAndSaveNicknameFor(
-        namer.getFullyQualifiedApiWrapperClassName(
-            context.getInterface(), context.getApiConfig().getPackageName()));
-
-    return InitCodeView.newBuilder()
-        .lines(generateSurfaceInitCodeLines(context, initCode))
-        .fieldSettings(getFieldSettings(context, initCode.getArgFields()))
-        .imports(importTypeTransformer.generateImports(typeTable.getImports()))
-        .apiFileName(
-            namer.getServiceFileName(
-                context.getInterface(), context.getApiConfig().getPackageName()))
-        .build();
+  public InitCodeView generateSmokeTestInitCode(
+      MethodTransformerContext context, SymbolTable table) {
+    SmokeTestConfig testConfig = context.getInterfaceConfig().getSmokeTestConfig();
+    InitCodeNode rootNode =
+        InitCodeNode.createSpecItemTree(
+            SpecItemParserContext.newBuilder()
+                .table(table)
+                .rootObjectType(testConfig.getMethod().getInputType())
+                .initValueConfigMap(createInitValueMap(context))
+                .sampleCodeInitFields(testConfig.getInitFields())
```

Comment:
I am not sure if sampleCodeInitFields is the best name here. Since it is actually used by both sample and test gen